src/input/controller_adapter_system.cpp

```diff
@@ -0,0 +1,228 @@
+};
+}
+
+ControllerAdapterSystem::ControllerAdapterSystem()
+    : mScene( Scene::Get() )
+    , mCalibrate( false )
+{
+}
+
+
+void ControllerAdapterSystem::Init()
+{
+}
+
+
+void ControllerAdapterSystem::Update(double DeltaTime)
+{
+    Opt<InputSystem> inputsys = InputSystem::Get();
+    if( !inputsys.IsValid() )
+    {
+        return;
+    }
+
+    // collect controlled players by controller id
+    std::map<int32_t, Opt<Actor> > controlledPlayers;
+    static core::ProgramState& ps =  core::ProgramState::Get();
+    Opt<Actor> actor( mScene.GetActor( ps.mControlledActorGUID ) );
+    Opt<core::ClientData> clientData( ps.FindClientDataByActorGUID( ps.mControlledActorGUID ) );
+    if( clientData.IsValid() && actor.IsValid() )
+    {
+        controlledPlayers[ clientData->mControlledLocalPlayerId ] = actor;
+    }
+
+    for( auto joy : getJoysticks() )
+    {
+        // get keymapping for joy name @ joy pos ( if custom, support per-slot mapping )
+        ControllerState cs( joy );
+        if( !cs.isValid() )
+        {
+            continue;
+        }
+
+        if( mCalibrate )
+        {
+            std::cout << "Joy " << joy << "\n";
+            std::cout << "\tname: '" << cs.name << "'\n";
+            std::cout << "\taxes: \n";
+            int cnt = 0;
+            for( auto const& a : cs.axes )
+            {
+                std::cout << "\t\ta" << cnt++ << " " << a << "\n";
+            }
+            std::cout << "\tbuttons: \n";
+            cnt = 0;
+            for( auto const& b : cs.buttons )
+            {
+                std::cout << "\t\tb" << cnt++ << " " << b << "\n";
+            }
+        }
+
+        // get controlled player for joy
+        int32_t controlledLocalPlayerId = 1;
+        static input::PlayerControlDevice& pcd( input::PlayerControlDevice::Get() );
+        if( pcd.GetControlDevice( controlledLocalPlayerId ) != input::PlayerControlDevice::Controller ||
+            pcd.GetControllerIndex( controlledLocalPlayerId ) != joy )
+        {
+            continue;
+        }
+
+        // get InputState for player
+        InputState is = inputsys->GetInputState( controlledLocalPlayerId );
+
+        // fill inputstate for player
+        is.mShoot = cs.getBool( "shoot" );
+        is.mShootAlt = cs.getBool( "shoot_alt" );
+        is.mUseNormalItem = cs.getBool( "use_normal" );
+        is.mReload = cs.getBool( "reload" );
+        is.mShowLeaderboard = cs.getBool( "show_leaderboard" );
+        is.mPause = cs.getBool( "pause" );
+        double vh = cs.getDouble( "view_horizontal" ), vv = cs.getDouble( "view_vertical" );
+        if( std::abs( vv ) >= 0.4 || std::abs( vh ) >= 0.4 )
+        {
+            is.mOrientation = atan2( vv, vh );
+        }
+        Opt<Actor> actor = controlledPlayers[ controlledLocalPlayerId ];
+        if( actor.IsValid() )
+        {
+            Opt<IPositionComponent> actorPositionC = actor->Get<IPositionComponent>();
+            static double const radius = 400;
```

Comment:
Donezo, for now all 3 (view_null_zone, move_null_zone, cursor_radius) under settings/controllers